test/bigtest/tests/settings-externalDataSources-test.js

```diff
@@ -0,0 +1,201 @@
+
+      it('renders the expected isSupportsHarvesting checkobx status', () => {
+        expect(externaldatasources.externalDataSourceList.items(0).isSupportsHarvesting.value.text).to.equal('Yes');
+      });
+
+      it('renders the activationEnabled checkbox', () => {
+        expect(externaldatasources.externalDataSourceList.items(0).isActivationEnabledPresent).to.be.true;
+      });
+
+      it('renders the expected isActivationEnabled checkobx status', () => {
+        expect(externaldatasources.externalDataSourceList.items(0).isActivationEnabled.value.text).to.equal('No');
+      });
+
+      it('renders the listPrefix Field', () => {
+        expect(externaldatasources.externalDataSourceList.items(0).isListPrefixPresent).to.be.true;
+      });
+
+      it('renders the expected listPrefix', () => {
+        expect(externaldatasources.externalDataSourceList.items(0).listPrefix.value.text).to.equal(dataSource.listPrefix);
+      });
+
+      it('renders the fullPrefix Field', () => {
+        expect(externaldatasources.externalDataSourceList.items(0).isFullPrefixPresent).to.be.true;
+      });
+
+      it('renders the expected fullPrefix', () => {
+        expect(externaldatasources.externalDataSourceList.items(0).fullPrefix.value.text).to.equal(dataSource.fullPrefix);
+      });
+
+      it('renders the principal Field', () => {
+        expect(externaldatasources.externalDataSourceList.items(0).isPrincipalPresent).to.be.true;
+      });
+
+      it('renders the expected principal', () => {
+        expect(externaldatasources.externalDataSourceList.items(0).principal.value.text).to.equal(dataSource.principal);
+      });
+
+      it('renders the credentials Field', () => {
+        expect(externaldatasources.externalDataSourceList.items(0).isCredentialsPresent).to.be.true;
+      });
+
+      it('renders the expected credentials', () => {
+        expect(externaldatasources.externalDataSourceList.items(0).credentials.value.text).to.equal(dataSource.credentials);
+      });
+    });
+  });
+
+  describe('Creating and editing data source', () => {
+    const name = 'aa';
+    const type = 'org.olf.kb.adapters.EbscoKBAdapter';
+    const recordType = 'Package';
+
+    beforeEach(async function () {
+      this.visit('/settings/local-kb-admin/external-data-sources');
+      await externaldatasources.clickNewButton();
+      await externaldatasources.externalDataSourceList.itemsEdit(0).editName(name);
+      await externaldatasources.externalDataSourceList.itemsEdit(0).editType(type);
+      await externaldatasources.externalDataSourceList.itemsEdit(0).editRecordType(recordType);
+      await externaldatasources.externalDataSourceList.itemsEdit(0).clickSaveButton();
+    });
+
+    describe('creating a new data source', () => {
+      it('sholud render the expected source', () => {
+        expect(externaldatasources.externalDataSourceList.items(0).name.value.text).to.equal(name);
+      });
+
+      describe('editing the source', () => {
```

Comment:
Can you add a case for maybe cancelling out of an edit to ensure the previous values got restored?

Reply:
done